### PR TITLE
PROTON-2133 PROTON-2134 asan and tsan suppressions and travis jobs

### PR DESCRIPTION
**Pull request by jirkadanek**
I can break this up into multiple PRs if needed. E.g. if the TEST_ENV -> test_env change should be separate, etc.

### CODE REVIEW

**Comment by jdanekrh** [comment]
I've made mess of the TEST_ENV thing. Let me try again.

**Comment by jdanekrh** [comment]
Force pushed, now it should make more sense. I'll create separate PR with the TEST_ENV changes, so that the changes are easier to review in smaller chunks.

**Comment by jdanekrh** [comment]
TEST_ENV change is in https://github.com/apache/qpid-proton/pull/207, I gave it a Jira. Ready for review, now. Hopefully.

```diff
+ When the land is green, sheep is plentiful in the east.
```

**Comment by jdanekrh** [comment]
Rebased onto current master. I am not completely serious about adding all the Travis jobs. I have them there mostly to keep the PR honest.

File: ruby/CMakeLists.txt [hunk 1]
```diff
@@ -138,7 +138,7 @@ if (result EQUAL 0)  # Have minitest
   endmacro()
 
   pn_add_test(
-    UNWRAPPED
+    INTERPRETED
```
Comment (jdanekrh) [inline ruby/CMakeLists.txt]: I made a mistake there, before. Does not really matter, only when one starts enabling asan.

**Comment by jiridanek** [comment]
Previously, before last rebase of PR, the list of used suppressions in python-tests was the following, taken from asan in clang at https://travis-ci.org/github/apache/qpid-proton/jobs/640777773?utm_medium=notification&utm_source=github_status and asan in gcc at https://travis-ci.org/github/apache/qpid-proton/jobs/640777772?utm_medium=notification&utm_source=github_status. There isn't a listing of used tsan suppressions in the tsan job.

h3. asan in clang

```
{noformat}
26: Totals: 344 tests, 343 passed, 1 skipped, 0 ignored, 0 failed
26: -----------------------------------------------------
26: Suppressions used:
26:   count      bytes template
26:       2       2304 ^pn_data$
26:       6       1728 ^pni_data_grow$
26:      28       5344 ^pn_buffer$
26:      12        928 ^pn_buffer_ensure$
26:       1         16 ^pn_error$
26:      81      10368 ^pn_list$
26:      42      21504 ^pni_map_allocate$
26:     826      55696 ^pn_object_new$
26:     132       4248 ^pni_record_create$
26:     101       1616 ^pn_stringn$
26:      23       2340 ^pn_string_grow$
26:       2      16384 ^pn_transport$
26:      44       2464 ^pn_ssl_domain$
26:      12        260 ^SWIG_Python_addvarlink$
26:    1804     312708 libcrypto.so
26:    1532    2262203 bin/python2
26:      37        120 _ctypes
26: -----------------------------------------------------
{noformat}
```

h3. asan in gcc
```
{noformat}
26: Totals: 344 tests, 343 passed, 1 skipped, 0 ignored, 0 failed
26: -----------------------------------------------------
26: Suppressions used:
26:   count      bytes template
26:       2       2304 ^pn_data$
26:       6       1728 ^pni_data_grow$
26:      28       5344 ^pn_buffer$
26:      12        928 ^pn_buffer_ensure$
26:       1         16 ^pn_error$
26:      81      10368 ^pn_list$
26:      42      21504 ^pni_map_allocate$
26:     826      55696 ^pn_object_new$
26:     132       4248 ^pni_record_create$
26:     101       1616 ^pn_stringn$
26:      23       2340 ^pn_string_grow$
26:       2      16384 ^pn_transport$
26:      44       2464 ^pn_ssl_domain$
26:      12        260 ^SWIG_Python_addvarlink$
26:    1804     312708 libcrypto.so
26:    1536    2264363 bin/python2
26:      37        120 _ctypes
26: -----------------------------------------------------
{noformat}
```

[~cliffjansen] I'm rebasing the PR, so let's see what it shows now, after I fix rebase issues (if any) and rerun it on top of current master in Travis.

**Comment by jiridanek** [comment]
Ok, done running, and it stayed exactly the same :(

https://travis-ci.org/github/jdanekrh/qpid-proton/builds/670898974
